src/core/thread/mesh_forwarder.cpp

```diff
@@ -999,6 +999,36 @@ void MeshForwarder::HandleSentFrame(Mac::Frame &aFrame)
 
     aFrame.GetDstAddr(macDest);
 
+    if ((neighbor = mMle.GetNeighbor(macDest)) != NULL)
+    {
+        switch (aStatus)
+        {
+        case Mac::Sender::kStatusSuccess:
+            neighbor->mLinkFailures = 0;
+            break;
+
+        case Mac::Sender::kStatusNoAck:
+            neighbor->mLinkFailures++;
+
+            if (mMle.IsActiveRouter(neighbor->mValid.mRloc16))
+            {
+                if (neighbor->mLinkFailures >= Mle::kFailedRouterTransmissions)
+                {
+                    mMle.RemoveNeighbor(*neighbor);
+                }
+            }
+            else
+            {
+                if (neighbor->mLinkFailures >= Mle::kFailedChildTransmissions)
+                {
+                    mMle.RemoveNeighbor(*neighbor);
```

Comment:
nit: Spec defines in 4.7.5 a little bit different approach when FAILED_CHILD_TRANSMISSION is achieved for the child. We may consider to create a new state like Neighbor::kStateLinkBroken, to retain actual child's data and perform full removal only when timeout occurs. However it may be done in the future.